### DISCUSSION TITLE
Battery Level LED with RPCS3

### DISCUSSION THREAD

**Discussion by HendyStark** [nefarius/DsHidMini#185] [Support]
First, i want to say thank you very much for making this, DsHidMini is the reason for me to buy a Dualshock 3 to play emulators, before this i only play with mouse and keyboard. I just recieved it 10 hours ago, and it just works with DsHidMini, cool.

Just now i noticed something with the led, here's what i do :
1. Charged the DS3 until it's full, unplug, and the the number 4 LED is turned on just like what the Feature section of the DsHidMini github page says.
2. I play Duckstation, the LED stiil on the number 4, normal.
3. Then i play RPCS3, when i start the emulator, the LED still on number 4, then i start the game, the LED that's on now is number 1.
4. I close RPCS3 and go back to Duckstation, the LED still stuck at number 1, not on number 4 because it's still fully charged.

The only thing i can do after i played RPCS3 so the LED go back to number 4 is to turn off and then turn on the DS3. There was also another problem i encountered, after i played RPCS3, i restarted the DS3 so the LED is back on 4, but instead all the LED are off when playing Duckstation or other emulator that's not RPCS3, when i played RPCS3 number 1 is on again, the only thing i can do with this one so that the LED are working again is using the reset button on the back of DS3.

Is this an expected behavior with RPCS3?
Sorry if my writing is weird, english is not my first language, once again, thank you!

**Comment by nefarius** [comment]
Hi!

> Is this an expected behavior with RPCS3?

Yes, this is expected behaviour. Once you run DSHM with RPCS3 the driver will "hand over" control over LED and rumble to RPCS3 so if it decides to alter the LED, that happens. Since the driver has no way of knowing that a new app is accessing it afterwards, the LED will remain at the last state it was instructed to by RPCS3.

[It's also briefly covered in the docs](https://vigem.org/projects/DsHidMini/HID-Device-Modes-Explained/#sxs).

Cheers

**Reply by HendyStark** [reply]
> Hi!
> 
> > Is this an expected behavior with RPCS3?
> 
> Yes, this is expected behaviour. Once you run DSHM with RPCS3 the driver will "hand over" control over LED and rumble to RPCS3 so if it decides to alter the LED, that happens. Since the driver has no way of knowing that a new app is accessing it afterwards, the LED will remain at the last state it was instructed to by RPCS3.
> 
> [It's also briefly covered in the docs](https://vigem.org/projects/DsHidMini/HID-Device-Modes-Explained/#sxs).
> 
> Cheers

Oh, seems like i missed that one lol, sorry about that, i should've read everything more carefully. Thank you very much!

**Reply by nefarius** [reply]
No worries, glad you enjoy the work! ❤️
